client/src/app/file-upload/file-upload.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit, Input, ViewChild, OnChanges, SimpleChanges, ElementRef } from '@angular/core';
+import { InvoiceImageService } from '../services/invoice-image.service';
+import { InvoiceImage } from '../model/InvoiceImage';
+
+@Component({
+  selector: 'app-file-upload',
+  templateUrl: './file-upload.component.html',
+  styleUrls: ['./file-upload.component.css']
+})
+
+
+export class FileUploadComponent implements OnInit, OnChanges {
+  @Input() requestIdInput: number;
+  @ViewChild('file') file;
+  @ViewChild('downloadLink') downloadLink: ElementRef;
+
+  images: [number, string][];
+  uploadingImages: uploadingImage[] = [];
+  requestId: number;
+  fileIsTooLarge : boolean = false;
+
+  constructor(private invoiceImageService: InvoiceImageService) { }
+
+  ngOnInit() {  }
+
+  ngOnChanges(changes: SimpleChanges) {
+
+    if (!changes.requestIdInput || changes.requestIdInput.currentValue == undefined) {
```

Comment:
Do this using observables in ngOnInit

Reply:
During ngOnInit requestIdInput is not yet initialized.